adapter/internal/operator/synchronizer/data_store.go

```diff
@@ -86,6 +86,10 @@ func (ods *OperatorDataStore) processAPIState(apiNamespacedName types.Namespaced
 		}
 	} else {
 		cachedAPI.ProdHTTPRoute = nil
+		if cachedAPI.ProdHTTPRoute != nil {	
```

Comment:
this line always would be false due to above line

Reply:
fixed the logic.